File: Flutter-Learning-English-Server/models/Word.js
const mongoose = require('mongoose')
const Schema = mongoose.Schema

const WordSchema = new Schema({
    username: {
        type: String,
        require: true
    },
    topicId: {
        type: Schema.Types.ObjectId,
        ref: 'Topic',
        require: true
    },
    english: {
        type: String,
        require: true
    },
    vietnamese: {
        type: String,
        require: true
    },
    description: String,
    isStarred: {
        type: Boolean,
        default: false
    },
    numberCorrect: {
        type: Number,
        default: 0
    },
    status: {
        type: String,
        enum: ['not learned', 'currently learning', 'mastered'],
        default: 'not learned'
    },
    copyId: {
        type: Schema.Types.ObjectId,
        ref: 'Word'
    }
})

module.exports = mongoose.model('Word', WordSchema)

File: Flutter-Learning-English-Application/lib/models/word.dart
class Word {
  final String id;
  final String username;
  final String topicId;
  final String english;
  final String vietnamese;
  final String description;
  late final bool isStarred;
  final int numberCorrect;
  final String status;

  Word({
    required this.id,
    required this.username,
    required this.topicId,
    required this.english,
    required this.vietnamese,
    required this.description,
    required this.isStarred,
    required this.numberCorrect,
    required this.status,
  });

  factory Word.fromJson(Map<String, dynamic> json) {
    return Word(
      id: json['_id'],
      username: json['username'],
      topicId: json['topicId'],
      english: json['english'],
      vietnamese: json['vietnamese'],
      description: json['description'] ?? '',
      isStarred: json['isStarred'],
      numberCorrect: json['numberCorrect'],
      status: json['status'],
    );
  }
}
